cmd/minikube/cmd/cp.go

```diff
@@ -83,6 +88,43 @@ Example Command : "minikube cp a.txt /home/docker/b.txt" +
 func init() {
 }
 
+// deriveDstPath sets the src filename as dst filename
+// when the dst file name is not provided and ends with a `/`.
+// Otherwise this function is a no-op and returns the passed dst.
+func deriveDstPath(dst, src string) string {
+	dd, df := filepath.Split(dst)
+
+	// if dst is empty, dd and df will be empty, so return dst
+	// validation will be happening in `validateArgs`
+	if dd == "" && df == "" {
+		return ""
+	}
+
+	// if filename is already provided, return dst
+	if df != "" {
+		return dst
+	}
+
+	_, sf := filepath.Split(src)
+
+	// if src filename is empty, return dst
+	if sf == "" {
+		return dst
+	}
+
+	// Don't use os specific Join() when coping from host
+	// to Minikube node (Linux)
+	// So if [src] is host and [dst] is Minikube, use path.Join()
+	// Ref: https://github.com/kubernetes/minikube/pull/15519/files#r1261683815
+	if newRemotePath(src).node == "" {
+		return path.Join(dd, sf)
+	}
+
+	// if [dst] filename is not provided and [src] filename is provided,
+	// join [dst] dir and [src] filename
+	return filepath.Join(dd, sf)
```

Comment:
This fails on the case that you're copying from one node to another as `src.node != ""` so will do `filepath.Join`.

Here's a table for all the cases
```
cp host guest                      => path
cp host minikube:guest             => path
cp minikube:guest host             => filepath
cp minikube1:guest minikube2:guest => path
```

So it should be easiest to check if `src.node != "" && dst.node == ""` then use filepath, for everything else can use path.

Reply:
Fixed the above case + the following cases

```
cp host guest                      => path
    src: filepath.split
    dst: path.split 
cp host minikube:guest             => path
    src: filepath.split
    dst: path.split 
cp minikube:guest host             => filepath
    src: path.split
    dst: filepath.split 
cp minikube1:guest minikube2:guest => path
    src: path.split
    dst: path.split 
```